### PR TITLE
added condition for distributed logic

### LINKED ISSUE #3862: Distributed defaults to using ListSortedGameServersPriorities when CountsAndLists is enabled
**What happened**:

Counters and Lists moved to Beta, so it's now on by default. If the scheduling type is set to `Distributed` it will execute the logic for CountersAndLists instead of the logic for `Distributed`. https://github.com/googleforgames/agones/blob/aa390f9f83564b04db9abd6dc985ca106cfa286f/pkg/gameserverallocations/find.go#L54

**What you expected to happen**:

If there are Counters and Lists Priorities https://github.com/googleforgames/agones/blob/aa390f9f83564b04db9abd6dc985ca106cfa286f/pkg/apis/agones/v1/fleet.go#L83, the allocator should use the Counters and Lists logic. Otherwise, the Allocator should use the `Distributed` logic.

### PR DESCRIPTION
**Pull request by ashutosji**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:https://github.com/googleforgames/agones/issues/3862

Closes #3862 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/gameserverallocations/find.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ func findGameServerForAllocation(gsa *allocationv1.GameServerAllocation, list []
 
 	var loop func(list []*agonesv1.GameServer, f func(i int, gs *agonesv1.GameServer))
 
+	// Determine if Priorities are set in the Fleet spec
+	prioritiesSet := len(gsa.Spec.Priorities) > 0
```
Comment (igooch) [inline pkg/gameserverallocations/find.go:44]: This can be moved to inside the `case apis.Distributed:`

**Review by igooch — COMMENTED** [review]
Looks good, just one nit.

File: pkg/gameserverallocations/find.go [hunk 2]
```diff
@@ -49,9 +49,12 @@ func findGameServerForAllocation(gsa *allocationv1.GameServerAllocation, list []
 			}
 		}
 	case apis.Distributed:
+		// Determine if Priorities are set in the Fleet spec
+		prioritiesSet := len(gsa.Spec.Priorities) > 0
+
 		// randomised looping - make a list of indices, and then randomise them
 		// as we don't want to change the order of the gameserver slice
-		if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) || !prioritiesSet {
```
Comment (markmandel) [inline pkg/gameserverallocations/find.go:57]:
```suggestion
		if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) || len(gsa.Spec.Priorities) == 0 {
```

Small thing, but probably doesn't need a variable? WDYT @igooch ?

Comment (igooch) [reply pkg/gameserverallocations/find.go:57]: No strong feelings either way. It's probably not needed.

Comment (markmandel) [reply pkg/gameserverallocations/find.go:57]: @ashutosji bump on removing this variable, and just moving the check inline, and then I think this is good to merge.

Comment (ashutosji) [reply pkg/gameserverallocations/find.go]: > @ashutosji bump on removing this variable, and just moving the check inline, and then I think this is good to merge.

Thanks! Moved the check inline.

**Review by igooch — APPROVED** [review]
LGTM

**Pull request merged**